File: js_nn/flappyBird_pixel_input/ga.js
function nextGen() {
    calculateFitness();
    for(let i = 0; i< total_pop; i++){
        birds[i] = pickOne();
    }
    console.log("next generation");
    saved_birds = [];
}

function pickOne(list, prob){
     var index = 0;
     var r = random(1);

     while (r > 0){
         r = r-saved_birds[index].fitness;
         index++;
     }
    index--;
    let bird = saved_birds[index];
    let child = new Bird(bird.brain);
    child.mutate();

    return child;}

function calculateFitness() {
    let sum = 0;
    for(let bird of saved_birds) {
        sum += bird.score;
    }
    for(let bird of saved_birds) {
    bird.fitness = bird.score / sum;
   }
}

File: python_nn/nn/ga.py
import random

import bird


def nextGen(savedBirds):
    #calculate fitness of savedBirds
    savedBirds = calculateFitness(savedBirds)


def pickOne(savedBirds):
    index = 0
    r = random.random()

    while( r > 0 ):
        r = r-savedBirds[index].fitness
        index += 1
    index -= 1
    bird_select = savedBirds[index]

    child = bird(bird_select.brain)
    child.mutate()
    return child

def calculateFitness(savedBirds):
    #return array with saved birds
    sum = 0
    for bird in savedBirds:
        sum += bird.score
    for bird in savedBirds:
        bird.fitness = bird.score / sum

    return savedBirds